### ISSUE TITLE
Core ESLint rule `no-useless-assignment` wrongly reports in Vue SFCs

### ISSUE THREAD

**Issue by alphatr** [vuejs/eslint-plugin-vue#2660]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 9.17.0
- **eslint-plugin-vue version:** 9.32.0
- **Vue version:** 3.5.13
- **Node version:** v22.11.0
- **Operating System:** macos 15.1.1

**Please show your full configuration:**

```json5
import pluginVue from 'eslint-plugin-vue'

export default [
  ...pluginVue.configs['flat/base'],
  {
    rules: {
      'no-useless-assignment': 'error',
    }
  }
]
```

**What did you do?**	

```vue
<template>
  <p :title="title">test</p>
</template>
<script setup>
import {ref} from "vue"

const title = ref("test")
const color = ref("#FFF")
</script>
<style>
p {
  color: v-bind(color);
}
</style>
```

**What did you expect to happen?**

doesn't report error;

**What actually happened?**

report error 'This assigned value is not used in subsequent statements.'

**Repository to reproduce this issue**

[https://ota-meshi.github.io/eslint-plugin-vue-demo](https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJxNjssKwjAQRX9lGDcKloLLGF32K7LROkognYTMtCCl/24aH7gaLuc+ZsY+3ggNWqUhhYvS2TGATWDUa6CTw3odnpVEbZsKt+2f2UqffVIQ0nGFfkgxK8yZ7gvccxzA4TSWAseO+8iiUBvhBMWyLf2l1+HuC/sYYv7BTdd1Fdr2vVMX9RnW6QTz+mxNGJiaq+fbtqrd0fFSQx8r7jGPgQTNjBybUagIaS4i/sEDsaI5LMsLNNpfGQ==)

**Comment by waynzh** [comment]
I can confirm this issue. 
It's likely something in `vue-eslint-parser` or we might need to add `vue/no-useless-assignment` to handle it. I haven't looked too closely yet, but will check when I have time.

**Comment by aparajita** [comment]
Will also flag the assigned to `foo` with something like this:

```vue
<script setup lang="ts">
  const foo = 'bar'
</script>
<template>
  <p>{{ foo }}</p>
</template>
```

**Comment by JoostKersjes** [comment]
As of ESLint v10, the `eslint:recommended` config enables `no-useless-assignment` by default. In case anyone lands on this issue like me, you can disable this rule for Vue SFC's like so:

```js
  {
    files: ["**/*.vue"],
    rules: {
      "no-useless-assignment": "off",
    },
  },
```

**Comment by FloEdelmann** [comment]
Can you please check whether it works with [vue-eslint-parser v10.4.0](https://github.com/vuejs/vue-eslint-parser/releases/tag/v10.4.0)?

This should be fixed by 

- https://github.com/vuejs/vue-eslint-parser/pull/288

**Comment by JoostKersjes** [comment]
@FloEdelmann That fixed it for my two projects, thanks! I'm guessing `vue-eslint-parser` isn't meant to be a direct dependency for most projects?

**Comment by FloEdelmann** [comment]
It is a peer dependency of eslint-plugin-vue, to allow individual updates.

Anyway, thanks for checking, then this can be closed.
